apps/prairielearn/src/components/QuestionScore.html.ts

```diff
@@ -0,0 +1,464 @@
+        ? html`
+            <a
+              class="grade-rate-limit-popover btn btn-xs"
+              data-toggle="popover"
+              data-html="true"
+              data-content="This question limits the rate of submissions. Further grade allowed ${instance_question.allow_grade_interval} (as of the loading of this page)."
+              data-placement="auto"
+              data-trigger="focus"
+              data-container="body"
+              tabindex="0"
+            >
+              <i class="fa fa-hourglass-half" aria-hidden="true"></i>
+            </a>
+          `
+        : ''}
+    </span>
+  `;
+}
+
+export function QuestionAwardedPoints({
+  instanceQuestionId,
+  previousVariants,
+  currentVariantId,
+  urlPrefix,
+}: {
+  instanceQuestionId: string;
+  previousVariants?: VariantWithScore[] | null;
+  currentVariantId?: string;
+  urlPrefix: string;
+}) {
+  if (!previousVariants) return '';
+  const MAX_DISPLAYED_VARIANTS = 10;
+  const collapseClass = `variants-points-collapse-${instanceQuestionId}`;
+  const collapseButtonId = `variants-points-collapse-button-${instanceQuestionId}`;
+
+  return html`
+    ${previousVariants.length > MAX_DISPLAYED_VARIANTS
+      ? html`
+          <button
+            id="${collapseButtonId}"
+            class="bg-white text-body p-0 m-0 border-0 rounded-0"
+            aria-label="Show older variants"
+            onclick="
+                // show all the hidden variant score buttons
+                document.querySelectorAll('.${collapseClass}').forEach(e => e.style.display = '');
+                // hide the ... button that triggered the expansion
+                document.querySelectorAll('#${collapseButtonId}').forEach(e => e.style.display = 'none');
+            "
+          >
+            &ctdot;
+          </button>
+        `
+      : ''}
+    ${previousVariants.map(
+      (variant, index) => html`
+        <a
+          class="badge ${currentVariantId != null && idsEqual(variant.id, currentVariantId)
+            ? 'badge-info'
+            : 'badge-secondary'} ${collapseClass}"
+          ${index < previousVariants.length - MAX_DISPLAYED_VARIANTS ? 'style="display: none"' : ''}
+          href="${urlPrefix}/instance_question/${instanceQuestionId}/?variant_id=${variant.id}"
+        >
+          ${variant.open ? 'Open' : `${Math.floor(variant.max_submission_score * 100)}%`}
+          ${currentVariantId != null && idsEqual(variant.id, currentVariantId)
+            ? html`<span class="sr-only">(current)</span>`
+            : ''}
+        </a>
+      `,
+    )}
+  `;
+}
+
+export function InstanceQuestionPoints({
+  instance_question,
+  assessment_question,
+  component,
+}: {
+  instance_question: Pick<
+    InstanceQuestion,
+    'auto_points' | 'manual_points' | 'points' | 'status' | 'requires_manual_grading'
+  >;
+  assessment_question: Pick<
+    AssessmentQuestion,
+    'max_auto_points' | 'max_manual_points' | 'max_points'
+  >;
```

Comment:
There are some pages using this component that override some of the types of individual variables. The `Pick` options here are to ensure that at least the columns we need are still properly defined. Also, some pages for now combine values like `max_*_points` into the instance question object, so this simplifies the handling of these pages until they are transitioned to a stronger type.